tensorflow/python/ops/nn_ops.py

```diff
@@ -2193,7 +2215,8 @@ def dropout(x, keep_prob, noise_shape=None, seed=None, name=None):  # pylint: di
     if tensor_util.constant_value(keep_prob) == 1:
       return x
 
-    noise_shape = noise_shape if noise_shape is not None else array_ops.shape(x)
+    noise_shape = _get_noise_shape(x, noise_shape)
```

Comment:
Is it possible to add this to https://github.com/tensorflow/tensorflow/blob/8c1f3b3fcdc5f897dbce3b7315c1b7000069f9c4/tensorflow/python/layers/core.py#L302 too?

Reply:
Thanks @vrv. The `tensorflow/python/layers/core.py` has been updated.